src/main/java/org/testng/providers/GuiceBackedObjectProvider.java

```diff
@@ -0,0 +1,52 @@
+package org.testng.providers;
```

Comment:
I'd prefer to host it under the internal package

Reply:
`org.testng.providers` is meant to be exposed to users because that is where we house the `IObjectProvider` interface. The other classes are either ways package visible only.